src/cov/cov-report.c

```diff
@@ -284,51 +282,51 @@ static void cover_print_html_header(FILE *f, cover_report_ctx_t *ctx, bool top,
               "  </head>\n"
               "  <section>\n\n");
 
-   if (!top) {
-      fprintf(f, "<nav>\n"
-                 "<b>Hierarchy:</b><br>\n");
-      int offset = 0;
-
-      ident_t full_hier = s->hier;
-      ident_t curr_id;
-      ident_t curr_hier = NULL;
-      const char *link = "../index";
-      do {
-         curr_id = ident_walk_selected(&full_hier);
-         curr_hier = ident_prefix(curr_hier, curr_id, '.');
-         if (offset > 0)
-            link = istr(curr_hier);
-         if (curr_id)
-            fprintf(f, "<p style=\"margin-left: %dpx\"><a href=%s.html>%s</a></p>\n",
-                        offset * 10, link, istr(curr_id));
-         offset++;
-      } while (curr_id != NULL);
-      fprintf(f, "</nav>\n\n");
-   }
-
    fprintf(f, "<header>");
-   va_start(ap, title);
-   vfprintf(f, title, ap);
-   va_end(ap);
+   fprintf(f, COV_RPT_TITLE "\n");
    fprintf(f, "</header>\n\n");
+}
+
+static void cover_print_navigation_tree(FILE *f, cover_report_ctx_t *ctx, cover_scope_t *s, ...)
+{
+   fprintf(f, "<nav>\n"
+               "<b>Hierarchy:</b><br>\n");
+   int offset = 0;
+
+   ident_t full_hier = s->hier;
+   ident_t curr_id;
+   ident_t curr_hier = NULL;
+   const char *link = "../index";
+
+   do {
+      curr_id = ident_walk_selected(&full_hier);
+      curr_hier = ident_prefix(curr_hier, curr_id, '.');
+      if (offset > 0)
+         link = istr(curr_hier);
+      if (curr_id)
+         fprintf(f, "<p style=\"margin-left: %dpx\"><a href=%s.html>%s</a></p>\n",
+                     offset * 10, link, istr(curr_id));
+      offset++;
+   } while (curr_id != NULL);
+
+   fprintf(f, "</nav>\n\n");
+
+}
 
+static void cover_print_file_and_inst(FILE *f, cover_report_ctx_t *ctx, cover_scope_t *s)
+{
    fprintf(f, "<h2 style=\"margin-left: " MARGIN_LEFT ";\">\n");
-   if (!top)
-      fprintf(f, "   Instance:&nbsp;%s\n", istr(s->hier));
-   else
-      fprintf(f, "   Instance:");
+   fprintf(f, "   Instance:&nbsp;%s\n", istr(s->hier));
    fprintf(f, "</h2>\n\n");
 
    cover_file_t *src = cover_file(&(s->loc));
    fprintf(f, "<h2 style=\"margin-left: " MARGIN_LEFT ";\">\n");
-   if (!top)
-      fprintf(f, "   File:&nbsp; <a href=\"../../%s\">../../%s</a>\n",
+   fprintf(f, "   File:&nbsp; <a href=\"../../%s\">../../%s</a>\n",
```

Comment:
I get a segfault here if NVC can't find the source file. This can occur if e.g. the working directory is different from when the architecture was run.

Reply:
Should be fixed now.

Reply:
Thanks. I do see a bunch of warnings (around 2-3 per module) about the source file being null. Is that expected?

Reply:
Perhaps needs to be guarded with `if (!loc_invalid_p(&(s->loc)) {` to make sure it has a valid location. Every node should really have a location set though.

Reply:
The `loc_invalid_p` is called inside of `cover_file`. I have added a warning in `cover_report_scope` that fires when
`cover_file` returns NULL. Reporting is skipped for such scope, so I think it is good to let user know something is
missing from the generated report.

@sean-anderson-seco , I can't reproduce your case. I have tried to call the `nvc -c` from different folders that I used to run `nvc -a` with `covdb` path adapted accrodingly, but it works in my case. I tried deleting the original source file, and
the warning fires then.

Reply:
It must be that the `tree_t t` argument to `cover_push_scope()` doesn't have a location set. @sean-anderson-seco can you work out what kind of node it is from the scope name? 

Anyway I don't think it's caused by the changes here.